runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContent.java

```diff
@@ -590,6 +592,19 @@ public void loadAppFromManifest(String url, String data) {
         mIsLoaded = true;
     }
 
+    public void setOriginAccessWhitelist(String url, String[] patterns) {
+        if (mNativeContent == 0 || url == null || url.isEmpty()) return;
```

Comment:
LGTM. One hint, TextUtils.isEmpty(url) is equals to url == null || url.isEmpty().